lib/ui/painting/vertices.cc

```diff
@@ -44,48 +27,48 @@ void Vertices::RegisterNatives(tonic::DartLibraryNatives* natives) {
 }
 
 bool Vertices::init(Dart_Handle vertices_handle,
-                    SkVertices::VertexMode vertex_mode,
+                    DlVertexMode vertex_mode,
                     tonic::Float32List& positions,
                     tonic::Float32List& texture_coordinates,
                     tonic::Int32List& colors,
                     tonic::Uint16List& indices) {
   UIDartState::ThrowIfUIOperationsProhibited();
-  uint32_t builderFlags = 0;
+
+  if (!positions.data()) {
+    return false;
+  }
+
+  DlVertices::Builder::Flags flags;
   if (texture_coordinates.data()) {
-    builderFlags |= SkVertices::kHasTexCoords_BuilderFlag;
+    flags = flags | DlVertices::Builder::kHasTextureCoordinates;
   }
   if (colors.data()) {
-    builderFlags |= SkVertices::kHasColors_BuilderFlag;
+    flags = flags | DlVertices::Builder::kHasColors;
   }
+  DlVertices::Builder builder(vertex_mode, positions.num_elements() / 2, flags,
+                              indices.num_elements());
 
-  SkVertices::Builder builder(vertex_mode, positions.num_elements() / 2,
-                              indices.num_elements(), builderFlags);
-
-  if (!builder.isValid()) {
+  if (!builder.is_valid()) {
```

Comment:
release all the data here.

Reply:
I don't see Release calls in other drawing methods. Is there a specific reason to release them here?

Reply:
If I'm undertanding the stack trace correctly it looks like this is causing a dart allocation to happen before releasing the data. But if it turns out you're rejecting data differently then maybe not.

Reply:
Typed data has strange rules around it - even though tonic makes this look like "just a const reference", it's actually wrapping some Dart API that's challenging to use correctly. Once you acquire typed data, you must not call any other Dart_API that could cause an allocation to happen until you release that typed data.

Reply:
We do a C++ new in the builder constructor, but no specific Dart_API calls. The store calls copy out of the dart typed arrays into our C++ object. How is this any different from the code flow before I changed the method name from the skia style isValid to the Dl style "is_valid"?

Reply:
And which stack trace are you looking at?

Reply:
I'm wondering if the branch taken case here wasn't previously covered by a test, and so the issue wasn't getting hit. The at the top of the function is new, and so its `return false` is getting covered by a test, and is hitting the issue.

Reply:
Hitting what issue? All tests pass now. The test at the top is returning from the function which means that no additional allocations can happen. gradient.cc deals with all sorts of typed lists and makes all sorts of Dl/Sk allocations while doing so without any calls to Release. The difference there seems to be that the MakeRefcounted is called in the Create call which is called during construction rather than in the init*() calls.